raiden/tests/unit/storage/migrations/test_v20_to_v21.py

```diff
@@ -0,0 +1,108 @@
+import json
+from datetime import datetime
+from pathlib import Path
+from unittest.mock import patch
+
+from raiden.storage.migrations.v20_to_v21 import upgrade_v20_to_v21
+from raiden.storage.sqlite import SQLiteStorage
+from raiden.utils.upgrades import UpgradeManager, UpgradeRecord
+
+
+def setup_storage(db_path):
+    storage = SQLiteStorage(str(db_path))
+
+    state_changes_file = Path(__file__).parent / 'data/v20_statechanges.json'
+    state_changes_data = json.loads(state_changes_file.read_text())
+    for state_change_record in state_changes_data:
+        storage.write_state_change(
+            state_change=json.dumps(state_change_record[1]),
+            log_time=datetime.utcnow().isoformat(timespec='milliseconds'),
+        )
+
+    chain_state_data = Path(__file__).parent / 'data/v20_chainstate.json'
+    chain_state = chain_state_data.read_text()
+    cursor = storage.conn.cursor()
+    cursor.execute(
+        """
+        INSERT INTO state_snapshot(identifier, statechange_id, data)
+        VALUES(1, 1, ?)
+        """, (chain_state,),
+    )
+    storage.conn.commit()
+    return storage
+
+
+def assert_state_changes_are_transformed(storage: SQLiteStorage) -> None:
+    batch_query = storage.batch_query_state_changes(
+        batch_size=50,
+        filters=[
+            ('_type', 'raiden.transfer.state_change.ContractReceiveChannelNew'),
+        ],
+    )
+
+    for state_changes_batch in batch_query:
+        for state_change in state_changes_batch:
+            data = json.loads(state_change.data)
+            assert 'mediation_fee' in data['channel_state']
+            assert data['channel_state']['mediation_fee'] == '0'
+
+    batch_query = storage.batch_query_state_changes(
+        batch_size=50,
+        filters=[
+            ('_type', 'raiden.transfer.mediated_transfer.state_change.ActionInitInitiator'),
+        ],
+    )
+
+    for state_changes_batch in batch_query:
+        for state_change in state_changes_batch:
+            data = json.loads(state_change.data)
+
+            assert 'allocated_fee' in data['transfer']
+            assert data['transfer']['allocated_fee'] == '0'
+
+
+def assert_snapshots_are_transformed(storage: SQLiteStorage) -> None:
+    _, snapshot = storage.get_latest_state_snapshot()
+    assert snapshot is not None
+
+    snapshot = json.loads(snapshot)
+
+    # tokennetworkaddresses_to_paymentnetworkaddresses =
+    tn_to_pn = snapshot['tokennetworkaddresses_to_paymentnetworkaddresses']
+    for payment_network in snapshot['identifiers_to_paymentnetworks'].values():
+        for token_network in payment_network['tokennetworks']:
+            assert payment_network['address'] in tn_to_pn
```

Comment:
can you add proper assert failure messages to all asserts in this test? Makes reading the test code also rather easier.

Reply:
Will do